### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by alessandropellegrini**
Realign `develop` branch to what is currently in `release-2.0.0`.    
This is the ultimate step to realign the development process to the internal branching model.

### CODE REVIEW

**Review by readbeard — DISMISSED** [review]

File: src/communication/mpi.c [hunk 1]
```diff
@@ -132,73 +114,46 @@ void send_remote_msg(msg_t *msg){
...
-	bool found_msg = true;
+	MPI_Message mpi_msg;
+	int pending;
 
-	/* - `pending_msgs` and `MPI_Recv` need to be in the same critical section.
-	 *    I could start an MPI_Recv with an empty incoming queue.
-	 * - `MPI_Recv` and `insert_bottom_half` need to be in the same critical section.
-	 *    messages need to be inserted in arrival order into the BH
-	 */
 	if(!spin_trylock(&msgs_lock))
```
Comment (readbeard) [inline src/communication/mpi.c:123]:
```suggestion
	if(!spin_trylock(&msgs_lock))
```

i think we could remove this, we tried to run without the lock and everything was working

File: src/scheduler/scheduler.c [hunk 2]
```diff
@@ -506,7 +506,7 @@ void schedule(void) {
 
 #ifdef HAVE_CROSS_STATE
 	if(resume_execution && !is_blocked_state(LPS(lid)->state)) {
-		printf("ECS event is finished mark %llu !!!\n", LPS(lid)->wait_on_rendezvous);
+		printf("ECS event is finished at LP %d mark %llu !!!\n", lid_to_int(lid), LPS(lid)->wait_on_rendezvous);
```
Comment (readbeard) [inline src/scheduler/scheduler.c]:
```suggestion
		printf("ECS event is finished at LP %d mark %llu !!!\n", lid_to_int(lid), LPS(lid)->wait_on_rendezvous);
```
...removable

File: src/core/core.h [hunk 3]
```diff
@@ -180,14 +192,18 @@ typedef struct _msg_hdr_t {
 	// TODO: non serve davvero, togliere
 	int   			type;
 	unsigned long long	rendezvous_mark;	/// Unique identifier of the message, used for rendez-vous event
-	int			alloc_tid;
+	unsigned int		alloc_tid;
```
Comment (readbeard) [inline src/core/core.h:195]:
```suggestion
	unsigned int		alloc_tid;
```

Do we need to send it both in header and in actual msg?

File: README.md [hunk 4]
```diff
@@ -84,9 +66,17 @@ executable, which is the model code already linked with the ROOT-sim library.
...
 `./model --sequential --nprc <number of required LPs>`
 This allows to spot errors in the implementation more easily.
 
-Then, to run it in parallel, type
+Then, to run it in parallel, type    
 `./model --np <number of available Cores> --nprc <number of required LPs>`
+
+To run in a distributed environment, you can use standard MPI commands,
+such as:    
+`mpiexec -n 2 -f hosts ./model --np 2 --nprc 16`
```
Comment (readbeard) [inline README.md]: are we sure of `-f` ? The documentation says `--hostfile`

Comment (readbeard) [reply README.md]: ps: that's a good case in which no "oversubscription" on nodes occurs...maybe it's clearer if we write also `--map-by node`?
But it's not really important.

File: src/communication/mpi.c (hunk 1 above)
Comment (alessandropellegrini) [reply src/communication/mpi.c:123]: We tried as well, and it's actually crashing. We can remove the spinlock, but this depends on #102

File: src/scheduler/scheduler.c (hunk 2 above)
Comment (alessandropellegrini) [reply src/scheduler/scheduler.c]: Done, although your other PR has commented out that line :smile:

File: README.md (hunk 4 above)
Comment (alessandropellegrini) [reply README.md]: Correct!

File: src/core/core.h (hunk 3 above)
Comment (alessandropellegrini) [reply src/core/core.h:195]: We use this information to free a buffer of a positive message once it is annihilated, so it should be kept. The messages should be anyhow rearranged significantly. So far so good for me.

**Review by readbeard — APPROVED** [review]

**Pull request merged**